reconcile/ocm_upgrade_scheduler.py

```diff
@@ -177,12 +172,72 @@ def version_conditions_met(version, history, ocm_name, workloads, upgrade_condit
     return conditions_met
 
 
+def gates_to_agree(version_prefix: str, cluster: str, ocm: OCM) -> list[str]:
+    """Check via OCM if a version is agreed
+
+    Args:
+        version_prefix (string): major.minor version prefix
+        cluster (string)
+        ocm (OCM): used to fetch infos from OCM
+
+    Returns:
+        bool: true on missing agreement
+    """
+
+    need_agreement = []
+    gate_ids = [gate["id"] for gate in ocm.get_version_gates(version_prefix)]
+    agreements = [
+        agreement["version_gate"]["id"]
+        for agreement in ocm.get_version_agreement(cluster)
+    ]
+
+    for gate in gate_ids:
+        if gate not in agreements:
```

Comment:
This makes the loop quadratic. Maybe turn `agreements` into a `set`?

Reply:
Indeed, refactored the code